code/modules/research/xenoarchaeology/machinery/analysis_base.dm

```diff
@@ -218,3 +216,34 @@ obj/machinery/anomaly/proc/GetResultSpecifity(var/datum/geosample/scanned_sample
 			specifity = rand(0, 0.5)
 
 	return specifity
+
+/obj/machinery/anomaly/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null)
+	var/list/data[0]
+	data["max_temperature"] = XENOARCH_MAX_TEMP
+	data["safety_temperature"] = XENOARCH_SAFETY_TEMP
+	data["temperature"] = temperature
+
+	data["target_ticks"] = target_scan_ticks
+	data["scan_process"] = scan_process
+
+	data["beaker"] = !!held_container
+	if (held_container)
+		data["beaker_name"] = held_container.name
+		var/list/beaker_contents[0]
```

Comment:
Why

The fixed size notation always annoys me - the lists aren't fixed size and never will be. This is also identical to `= list()`, and I prefer that much more for readability.